pkg/operators/wasm/wasm.go

```diff
@@ -212,11 +228,15 @@ func (i *wasmOperatorInstance) init(
 	gadgetCtx operators.GadgetContext,
 	target oras.ReadOnlyTarget,
 	desc ocispec.Descriptor,
+	cache wazero.CompilationCache,
 ) error {
 	ctx := gadgetCtx.Context()
 	rtConfig := wazero.NewRuntimeConfig().
 		WithCloseOnContextDone(true).
 		WithMemoryLimitPages(256) // 16MB (64KB per page)
+	if cache != nil {
+		rtConfig = rtConfig.WithCompilationCache(cache)
+	}
```

Comment:
nit: I think you can safely pass nil to `WithCompilationCache`, so the conditional shouldn't be needed.

Reply:
Good catch! It can indeed handle it: https://github.com/tetratelabs/wazero/blob/main/runtime.go#L172!